File: Java/2.java
public class Carro {
    private String marca;
    private String modelo;
    private int ano;
    private int velocidade; 

    public Carro(String marca, String modelo, int ano) {
        this.marca = marca;
        this.modelo = modelo;
        this.ano = ano;
        this.velocidade = 0;
    }


    public String detalhes() {
        return "Marca: " + marca + ", Modelo: " + modelo + ", Ano: " + ano;
    }


    public String acelerar(int incremento) {
        if (incremento < 0) {
            return "O valor de incremento deve ser positivo.";
        }
        this.velocidade += incremento;
        return marca + " " + modelo + " acelerou. Velocidade atual: " + velocidade + " km/h";
    }


    public String frear(int decremento) {
        if (decremento < 0) {
            return "O valor de decremento deve ser positivo.";
        }
        this.velocidade = Math.max(0, velocidade - decremento);
        return marca + " " + modelo + " freou. Velocidade atual: " + velocidade + " km/h";
    }


    public int getVelocidade() {
        return this.velocidade;
    }

 
    public String exibirVelocidade() {
        return "Velocidade atual do " + marca + " " + modelo + ": " + velocidade + " km/h";
    }


    public static void main(String[] args) {
        Carro carroTeste = new Carro("Toyota", "Corolla", 2020);

        System.out.println(carroTeste.acelerar(50));
        System.out.println(carroTeste.frear(20));
        System.out.println(carroTeste.exibirVelocidade());
    }
}

File: Python/2.py
class Carro:
    def __init__(self, marca, modelo, ano):
        self.marca = marca
        self.modelo = modelo
        self.ano = ano
        self._velocidade = 0  


    def detalhes(self):
        return f"Marca: {self.marca}, Modelo: {self.modelo}, Ano: {self.ano}"


    def acelerar(self, incremento):
        if incremento < 0:
            return "O valor de incremento deve ser positivo."
        self._velocidade += incremento
        return f"{self.marca} {self.modelo} acelerou. Velocidade atual: {self._velocidade} km/h"

   
    def frear(self, decremento):
        if decremento < 0:
            return "O valor de decremento deve ser positivo."
        self._velocidade = max(0, self._velocidade - decremento)
        return f"{self.marca} {self.modelo} freou. Velocidade atual: {self._velocidade} km/h"


    @property
    def velocidade(self):
        return self._velocidade


    def exibir_velocidade(self):
        return f"Velocidade atual do {self.marca} {self.modelo}: {self.velocidade} km/h"



carro_teste = Carro("Toyota", "Corolla", 2020)

# Exibindo os resultados
print(carro_teste.acelerar(50))
print(carro_teste.frear(20))
print(carro_teste.exibir_velocidade())
